client/client.go

```diff
@@ -214,6 +202,32 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// WaitForInclusion blocks until the requested data has been verified with an inclusion proof.
```

Comment:
This function doesn't seem to be used or tested anywhere?

(Also, I'd have thought any possible users would probably already have a leaf in their hands, so maybe change the parameter to `leaf` not `data`)